### ISSUE TITLE
update_cells assumes list is ordered

### ISSUE THREAD

**Issue by mattlyon93** [burnash/gspread#647]
**Problem**
within Worksheet.update_cells(), the first line is 
`values_rect = cell_list_to_rect(cell_list)`

which in turn calls

```
def cell_list_to_rect(cell_list):
    if not cell_list:
        return []

    rows = defaultdict(lambda: {})

    row_offset = cell_list[0].row
    col_offset = cell_list[0].col
```

Here the row offset and column offset are set as the first item in the list. This will not be the case for an unordered list (for example if you have the largest 'A1' notation cell first). If the smallest 'A1' notation cell is not first in the list, it will cause the list of lists returned by this function to be incorrect.

**Proposed solution**

sort the cell list first, within Worksheet.update_cells():
`cell_list.sort(key=lambda x: (x.row, x.col), reverse=False)`

**Comment by Knio** [comment]
I also just ran into this issue.

Note that the proposed solution only works if there *is* a top-left-most cell in the input this. The comment lower in the function implies that it can handle missing values, so this is not always the case. A more general solution would be:

```python
row_offset = min(c.row for c in cell_list)
col_offset = min(c.col for c in cell_list)
```

**Comment by emlazzarin** [comment]
This issue is correct, and a solution is waiting in #613 for @burnash to approve or request revision.

**Comment by taewookim** [comment]
Is the solution already merged? still having this issue ( i pip installed last week)

**Comment by burnash** [comment]
Fixed in #613 / v3.3.0
